beacon_node/lighthouse_network/src/rpc/codec/ssz_snappy.rs

```diff
@@ -471,20 +476,29 @@ fn handle_rpc_request<T: EthSpec>(
         ))),
         SupportedProtocol::BlocksByRootV2 => Ok(Some(InboundRequest::BlocksByRoot(
             BlocksByRootRequest::V2(BlocksByRootRequestV2 {
-                block_roots: VariableList::from_ssz_bytes(decoded_buffer)?,
+                block_roots: RuntimeVariableList::from_ssz_bytes(
+                    decoded_buffer,
+                    fork_context.spec.max_request_blocks as usize,
```

Comment:
should this update at the fork?

Reply:
I wanted to do this for very long. 
The way we have it now is that if a new fork introduces new configs, then we simply use those parameters if the fork has been scheduled. This is based on the assumption that the config changes aren't drastic enough across forks. I like this way of using fork specific constants.

We can just pick the value from ForkContext directly without going into the spec.

Reply:
I think the spec should contain all these values and we should go via the spec. 

The codecs and the RPC handler for that matter don't have a concept of slot-clock, so they can't update these values on fork boundaries. I'm not sure its a big enough problem to introduce slot-clock information to handle these cases. A client re-start would correct it, provided the `spec` that is returned from a `ForkContext` is the correct fork, once-we've crossed the fork boundary

Reply:
In either case, its an improvement on the hardcoded limits we were using before.

Reply:
> The codecs and the RPC handler for that matter don't have a concept of slot-clock

They both already have the `ForkContext`, which has the necessary info via `current_fork`, so we don't actually need to weave a slot clock through these. So it ends up being pretty easy to add.  I get that it doesn't really "make sense" for the rpc handler and ssz codec's to have fork awareness, the protocol versioning (or `fork_context_bytes`) exists so we don't to worry about forks. But the type is actually being changed by this config update which *could arguably* warrant incrementing `beacon_blocks_by_range` to `v3`. But since the type change doesn't change encoding and decoding I get why no one cares, and I won't be the one to bring it up (:joy:). 

For now I've hard-coded the old values, so we'll have to update them to the deneb values in our first release across the fork, unless we want to go with the runtime switching.

> A client re-start would correct it, provided the spec that is returned from a ForkContext is the correct fork, once-we've crossed the fork boundary

Assuming we're moving the derived values to `ChainSpec`, we'll have to derive the pre and post deneb values and store both in `ChainSpec`, because we have no concept of "current fork" when instantiating `ChainSpec`.  So we'll have to leave it at the higher value and remember to update it in the first post-fork release.